src/main/java/jenkins/plugins/openstack/compute/JCloudsLauncher.java

```diff
@@ -14,14 +16,12 @@
 /*
  * This class as designed, is not supposed to be shared among multiple computers.
  */
-public class JCloudsLauncher extends ComputerLauncher {
+public class JCloudsLauncher extends DelegatingComputerLauncher {
 
     private static final Logger LOGGER = Logger.getLogger(JCloudsLauncher.class.getName());
 
-    /*package for testing*/ /*almost final*/ @Nonnull ComputerLauncher launcher;
-
     public JCloudsLauncher(@Nonnull ComputerLauncher launcher) {
-        this.launcher = launcher;
+        super(launcher);
     }
 
```

Comment:
Note: JCloudsLauncher's beforeDisconnect method (lines 41-48) delegates to afterDisconnect and not to beforeDisconnect.
I've not changed this functionality in this PR, and I've not noticed any bad effects from this, but I suspect that it isn't correct.